### ISSUE TITLE
Unable to compile

### ISSUE THREAD

**Issue by SuuperW** [Arisotura/melonDS#1346]
I am unable to compile after pulling the most recent updates in the master branch. I ended up having to uninstall then re-install MSYS2 due to some issues with keys, so everything here should be working on a brand-new clean install of MSYS2 following the instructions in the README and MSYS2 installation page.

The errors I am getting are:

src/ARMJIT_x64/ARMJIT_Compiler.cpp: In member function 'void ARMJIT::Compiler::LoadReg(int, Gen::X64Reg)':
src/ARMJIT_x64/ARMJIT_Compiler.cpp:450:59: error: 'reg' is not a constant expression
  450 |         MOV(32, R(nativeReg), MDisp(RCPU, offsetof(ARM, R[reg])));
src/ARMJIT_x64/ARMJIT_Compiler.cpp: In member function 'void ARMJIT::Compiler::SaveReg(int, Gen::X64Reg)':
src/ARMJIT_x64/ARMJIT_Compiler.cpp:457:41: error: 'reg' is not a constant expression
  457 |     MOV(32, MDisp(RCPU, offsetof(ARM, R[reg])), R(nativeReg));

**Comment by nadiaholmquist** [comment]
Which compiler are you using and what version? OS version/arch?

**Comment by RSDuck** [comment]
this issue was caused by the update to GCC 10 or 11 (I forgot which one), though I thought I fixed it already and based on the line numbers and what's supposed to be on that line, you're not using latest source.

**Comment by SuuperW** [comment]
@RSDuck You are correct, my code was not up to date. I'm not sure what happened there.
After (actually, this time) getting the most recent code, it does compile. However it does not run.

First error is about zlib1.dll. I have updated that, now it complains that libgraphite2.dll is not installed. Dependency Walker shows that there is also another missing required dll file: libmd4c.dll.

This is on Windows 7, btw.

**Comment by nadiaholmquist** [comment]
If you did not statically link it, you need to copy the needed DLLs to the directory of the executable if you want to run it outside the MSYS2 shell. The `msys-dist.sh` script in the tools directory should do that for you.

**Comment by SuuperW** [comment]
I did run msys-dist.sh and am attempting to run it from the dist directory.

**Comment by SuuperW** [comment]
I just made a static build, and that one runs.
